cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -224,11 +232,16 @@ getTxId (ShelleyTxBody era tx _) =
 --
 
 data TxIn = TxIn TxId TxIx
-  deriving (Eq, Ord, Show)
+  deriving (Eq, Generic, Ord, Show)
+
+deriving instance ToJSON TxIn
+instance ToJSONKey TxIn where
+  toJSONKey = toJSONKeyText (\txIn -> Text.pack $ show txIn)
```

Comment:
Using show here is pretty unpleasant. How about just concatenating the txid as a hash, with the index. Perhaps with a separator character between them (like we use in the CLI).

Reply:
Yep this was removed in subsequent commits